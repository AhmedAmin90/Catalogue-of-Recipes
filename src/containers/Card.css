.Card {
    width: 350px;
    height: 350px;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    cursor: pointer;
    position: relative;
    margin: 10px;
    border-radius: 10px;
}

.Card:hover {
    transform: rotate(5deg);
    transition: 2s ease;
}

.Card-shadow {
    background-color: #215330;
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 20%;
    opacity: 0.8;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.Card-name {
    font-size: 15px;
    color: #FCC93C;
    text-align: center;
}