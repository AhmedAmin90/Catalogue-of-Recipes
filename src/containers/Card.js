import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ img, name }) => (
  <div className="Card" style={{ backgroundImage: `url(${img})` }}>
    <div className="Card-shadow">
      <h1 className="Card-name">{name}</h1>
    </div>
  </div>
);

Card.defaultProps = {
  img: '',
  name: '',
};

Card.propTypes = {
  img: PropTypes.string,
  name: PropTypes.string,
};

export default Card;
