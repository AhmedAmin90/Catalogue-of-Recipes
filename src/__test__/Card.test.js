import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import store from '../index';
import Card from '../containers/Card';

test('renders correctly', () => {
  const tree = create(<Card name="test" img="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('rendered Main', () => {
  let renderedComponent;
  beforeEach(() => {
    renderedComponent = render(
      <Provider store={store}>
        <Router>
          <Card img="Test Url" name="Test Name" />
        </Router>
      </Provider>,
    );
  });

  it('has a div with class loading', () => {
    const { container } = renderedComponent;
    const appElement = container.querySelector('.Card');
    expect(appElement).toBeInTheDocument();
    expect(appElement).toHaveStyle('background-image: url(\'Test Url\')');
  });
});
