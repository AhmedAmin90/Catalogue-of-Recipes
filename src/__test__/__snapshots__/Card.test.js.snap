// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Card"
  style={
    Object {
      "backgroundImage": "url(test)",
    }
  }
>
  <div
    className="Card-shadow"
  >
    <h1
      className="Card-name"
    >
      test
    </h1>
  </div>
</div>
`;
